### ISSUE TITLE
Force username prompt

### ISSUE THREAD

**Issue by gnouts** [excalidraw/excalidraw#3177]
When sharing a board with a lot of collaborators, you end up with a lot of "unknown user".

It might be nice to have an option to force prompt username. For example, if username is not set already and you open the link, display a popup that prompt name and greyd out the board.

Even if the popup is not mandatory and you can discard it with a click, most user will blindly fill it and it would give a smoothier experience for everyone.

I'm thinking of two implementations :
- add a button next to the share link, that enable prompt username for the board.
- make the prompt a default behaviour when you create a collaborative board. (could be toggle on/off server side for selfhosted instances)

**Comment by ad1992** [comment]
I agree the `unknown user` is not a good experience and most of the time it happens unknowingly for fresh users because the name is not filled. 

Showing a prompt might affect the smooth experience which we have right now (just open the link and start collaborating) so instead, if we can focus/highlight the collab button and show some kind of indicator, maybe a tooltip that informs that "username is not filled and will be shown as the unknown user to collaborators" so it can catch the attention of the user as well and doesn't interfere with the flow. Thoughts?

**Comment by gnouts** [comment]
Another idea, generate username on the fly like GoogleDoc do with their "anonymous_animal" names. Make it visible to the user that their name is "anonymous_panda" so they can change it by clicking on it.

**Comment by ad1992** [comment]
I like the idea of generating random usernames on the fly. That way flow would also remain smoother.

**Comment by h7y** [comment]
Can I take this up? :)

**Comment by dwelle** [comment]
@h7y I have a module for generating this. We may wanna create an npm package for it.

Do we want to put this in core? My implem for this has ~4KB (just the generator part) and I need to use this in the host app as well. So if we do it in core, we must export the API so host apps don't need to bundle it 2x.

But ultimately, the collaborator list shouldn't really live in core either. /cc @ad1992

**Comment by ad1992** [comment]
@dwelle yes let's have the npm package for name generator and use it here.
